src/dotnetLearning/FactoryApp/Service/FactoryAppService.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Microsoft.Extensions.Options;
+using dotnetLearning.FactoryApp.Service.FacilityService;
+using dotnetLearning.FactoryApp.View;
+
+namespace dotnetLearning.FactoryApp.Service
+{
+    public class FactoryAppService(IFacilityService facilityService, IView view, IOptions<FacilityServiceOptions> options)
+    {
+        internal async Task RunAsync()
+        {
+            if (options.Value.FacilitiesJsonFilePath is not null)
+                await facilityService.DeserializeDataJson(options.Value.FacilitiesJsonFilePath);
+            else
+                throw new ArgumentNullException("Ошибка в файле конфигурации appsettings.json");
```

Comment:
Напомню, метод `string.IsNullOrEmpty` позволит сразу проверить не только на null, но и на пустую строку. Опять же вкусовщина, но можно немного упростить вложенность:

```suggestion
            if (string.IsNullOrEmpty(options.Value.FacilitiesJsonFilePath))
                throw new ArgumentNullException("Ошибка в файле конфигурации appsettings.json");

            await facilityService.DeserializeDataJson(options.Value.FacilitiesJsonFilePath);
```